src/subprocess-posix.cc

```diff
@@ -223,6 +261,26 @@ SubprocessSet::SubprocessSet() {
     Fatal("sigaction: %s", strerror(errno));
   if (sigaction(SIGHUP, &act, &old_hup_act_) < 0)
     Fatal("sigaction: %s", strerror(errno));
+
+  memset(&act, 0, sizeof(act));
+  act.sa_flags = SA_SIGINFO | SA_NOCLDSTOP;
+  act.sa_sigaction = SigChldHandler;
+  if (sigaction(SIGCHLD, &act, &old_chld_act_) < 0)
+    Fatal("sigaction: %s", strerror(errno));
+}
+
+void SubprocessSet::CheckConsoleProcessTerminated() {
+  if (!s_sigchld_received)
+    return;
+  for (vector<Subprocess*>::iterator i = running_.begin();
```

Comment:
we are now using C++14, so I suggest you use range-based loops with `auto` here as in:

```
for (auto it = running_.begin(); it != running_.end(); ++it) {
  ...
}
```

Reply:
I'm not a big fan of `auto`, as it tends to obscure types, but I can use it.

I will not place `++i` in the loop however because this loop erases elements and therefore `i` must only be increased when an erase didn't happen.